src/main/java/seedu/taassist/model/student/StudentModuleData.java

```diff
@@ -28,15 +28,20 @@ public class StudentModuleData implements Identity<StudentModuleData>, Comparabl
      */
     public StudentModuleData(ModuleClass moduleClass) {
         requireNonNull(moduleClass);
-        this.moduleClass = moduleClass;
+        /*
+         * The following line is to remove references to session from the module class.
+         * In a StudentModuleData, only the name of the module class is used. Also in the
+         * JSON file, only the name is saved.
+         */
+        this.moduleClass = new ModuleClass(moduleClass.getClassName());
```

Comment:
Might become an issue if the stored `moduleClass` is used to check for existing sessions in the module class. Should be ok for now though since that is not what we are doing.

Reply:
> Might become an issue if the stored moduleClass is used to check for existing sessions in the module class.

This is not the intended use case for this moduleClass. The real moduleClass object is in model and should be retrieved for the actual session list. This moduleClass is sort of a reuse of the class, otherwise it would be a string only.